### ISSUE TITLE
[Feature Request] Add ability to download all brews as .zip folder from account

### ISSUE THREAD

**Issue by Gazook89** [naturalcrit/homebrewery#1129]
I would like the ability to quickly download my brews to my computer.

1. Download all source code as text files
2. Download all rendered brews as PDFs  (less important, to me personally)

Allowing this will help with backing things up easily.  This came to mind as I consider linking my Google Drive account*, and the thought of backing up all of my brews by going into each one and copy/pasting all the text into separate files fills me with dread.  I have done this previously, but since I'm not sure which ones have since been revised I have to redo them all.

---
*side note, does linking to Google Drive create a new folder or something in my Drive, filled with text files?  I honestly thought I had linked already with Drive, but don't see anything to indicate as such in my Drive folders, so i'm not sure.  When I go to the 'log-in' screen it shows I'm logged in as my Homebrewery username...if I'm logged in with Google, woudl it should my google email instead?  This is way off topic here, sorry.

**Comment by calculuschild** [comment]
As for your side note:

Linking to Google Drive will create a "Homebrewery" folder on your drive with text files. It doesn't automatically copy over existing brews though, so the folder will be empty to start. I *think* moving it around will still work fine if you don't want it in the root of your Google Drive. You would need to manually open each brew, then click the google Drive icon, at which point the next save will copy your brew to your google Drive and then remove the existing copy on the Hombrewery servers. If you linked your google account to an existing homebrewery account, the username is your old homebrewery name. Similarly, if you created a new account directly with the "log in with google" button, it would have asked you to sign in with an existing homebrewery account or create a new one so it could be linked.

You can tell if you are logged in with google by trying to create a new brew. If you have a linked Google account, it will default to creating the brew on your google drive when you save it for the first time.

**Comment by Gazook89** [comment]
Ok, awesome.  I think what confused me most was that I clicked the G Drive icon in the editor, which "turned it on", but did nothing else...I didn't know I had to make an edit (type a little), which causes the auto-saver to trigger, and then it would create a copy.  Not until I made an edit did anything happen...and since I didn't know this, I sat staring at it for a few moments before thinking "this isn't working".  And then my understanding is that if I turn on this G Drive link with all my existing "how to" brews, that I'll have to go back to my reddit posts and update my share links, correct?  It's not really a problem as most of my Share links are in comments rather than being "link posts".

**Comment by calculuschild** [comment]
We can probably make it save right away when you click the g drive button. That might be more clear to users.

And yes, you will get a new link if you move it to the g drive.

**Comment by Gazook89** [comment]
Also, adding an additional text after the icon that says "Linked to Drive" when activated would be nice.  Or, in the element to the right that normally just says "Saved.", have that updated to say "Saved to [Drive icon] Drive".  

It could also be more clear that saving to Google Drive is brew-specific....that all of your brews are not automatically backed up, and you have to turn it on for each one.

**Comment by calculuschild** [comment]
Do you want to make a new issue with these other fixes? Auto-save when gDrive turned on, helper text, etc.? So it doesn't get buried in this separate issue?

**Comment by G-Ambatte** [comment]
I was playing with adding a "Download Source" button on the Share page, next to the original "Source" button (which I relabelled "View Source" for clarity). This allows the user to download the source of the current brew to their local computer as a plain text file.

It did increase the size of `server.js` to ~240 lines, so [my branch](https://github.com/G-Ambatte/homebrewery/commit/7fccb7e03ebbdc5e124a50c9a26bcb790be26f78) isn't passing Lint/CircleCI checks at the moment.

**Comment by ericscheid** [comment]
Also, with stubs for google drive brews #2114 ..

1. the share links don't stop working if you move the save to there,
2. the (pubished) gdrive brews now appear on your user page for other users

.. so there're little reason not to use that approach.

Moving multiple brews over to gdrive is still a slow and painful process though.

**Comment by ericscheid** [comment]
... and I just discovered my old javascript bookmarklet for downloading the current brew stopped working a while back (when we turned the `share` navItem link into a `share` drop-menu).

Got it working again. Will also update it for running on user page for batch downloading before posting.
